src/backport.ts

```diff
@@ -83,6 +83,27 @@ export class Backport {
       );
 
       const commitShas = await this.github.getCommits(mainpr);
+
+      let commitShasToCherryPick: string[];
+
+      if (mainpr.commits == 1) {
+        // if the PR only has one commit, we don't care if the PR was squashed or rebased
```

Comment:
this would not always be right, because a single commit could also be "squashed" (mostly to modified commit title and messages)

Reply:
agreed, we should take the `merge_commit_sha` when there is only one commit anyways. Adapting.